tests/test_photonphase.py

```diff
@@ -36,5 +37,40 @@ def test_result():
     photonphase.sys.stdout = saved_stdout
 
 
+parfile_nicer = os.path.join(datadir, "ngc300nicer.par")
+parfile_nicerbad = os.path.join(datadir, "ngc300nicernoTZR.par")
+eventfile_nicer = os.path.join(datadir, "ngc300nicer_bary.evt")
+
+
+@pytest.mark.skipif(
+    "DISPLAY" not in os.environ, reason="Needs an X server, xvfb counts"
+)
+def test_nicer_result():
+    "Check that barycentered NICER data is processed correctly."
+    saved_stdout, photonphase.sys.stdout = photonphase.sys.stdout, StringIO("_")
+    cmd = "{0} {1}".format(eventfile_nicer, parfile_nicer)
+    photonphase.main(cmd.split())
+    lines = photonphase.sys.stdout.getvalue()
+    v = 0.0
+    for l in lines.split("\n"):
+        if l.startswith("Htest"):
+            v = float(l.split()[2])
+    # Check that H-test is greater than 725
+    log.warning("V:\t%f" % v)
+    assert v > 200.0
+    photonphase.sys.stdout = saved_stdout
+
+
+@pytest.mark.skipif(
+    "DISPLAY" not in os.environ, reason="Needs an X server, xvfb counts"
+)
+def test_AbsPhase_exception():
+    "Verify that passing par file with no TZR* parameters raises exception"
+    with pytest.raises(ValueError):
+        saved_stdout, photonphase.sys.stdout = photonphase.sys.stdout, StringIO("_")
```

Comment:
please don't ever import sys through another module. It just gives you the same sys as import sys, and it means this fails mysteriously if the other module doesn't do "import sys". 

More generally this mess can be avoided by using pytest's "capsys" fixture - it captures stdout and you can read it out, and it restores stdout even if an exception occurs in your test.

Reply:
Ah, I don't know where this idiom came from. I just copied it from the other command line tool tests.